evm/vm/computation.py

```diff
@@ -104,6 +104,14 @@ def is_success(self):
     def is_error(self):
         return not self.is_success
 
+    @property
+    def should_burn_gas(self):
+        return self.error and self.error.burns_gas
```

Comment:
Can you adjust this to use the `is_error` property and move `self.error` to be private at `self._error`.  Realizing that as part of this work, `self._error` should be clearly marked private to discourage using it.

Reply:
on it